src/services/survey.ts

```diff
@@ -23,16 +25,83 @@ export const listSurveys = async (options = {}): Promise<SurveyResponse> => {
     ...options,
   });
 
+  const deserializedResponse = await deserialize(response);
+
   return {
-    surveys: response.data.map((survey: any) => parseSurvey(survey)),
+    surveys: deserializedResponse.data.map((survey: any) =>
+      parseSurvey(survey)
+    ),
     batchInfo: parseBatchInfo(response.meta),
   };
 };
 
+export const getSurveyDetail = async (surveyId: string) => {
+  const { accessToken } = getUserToken();
+
+  const response = await get(`/surveys/${surveyId}`, {
+    headers: { Authorization: `Bearer ${accessToken}` },
+  });
+
+  const deserializedResponse = await deserialize(response);
+
+  return parseSurveyDetail(deserializedResponse.data);
+};
+
 const parseSurvey = (surveyResponse: any): Survey => ({
   id: surveyResponse.id,
-  title: surveyResponse.attributes.title,
-  description: surveyResponse.attributes.description,
-  coverImageThumbnailUrl: surveyResponse.attributes.coverImageUrl,
-  coverImageUrl: `${surveyResponse.attributes.coverImageUrl}l`,
+  title: surveyResponse.title,
+  description: surveyResponse.description,
+  coverImageThumbnailUrl: surveyResponse.coverImageUrl,
+  coverImageUrl: `${surveyResponse.coverImageUrl}l`,
+});
+
+const parseSurveyDetail = (surveyDetailResponse: any): SurveyDetail => {
+  let surveyIntro, surveyOutro;
+
+  let questions = surveyDetailResponse.questions.map((question: any) =>
+    parseSurveyQuestion(question)
+  );
+
+  [[surveyIntro], questions] = extractQuestionType(questions, 'intro');
+  [[surveyOutro], questions] = extractQuestionType(questions, 'outro');
+
+  return {
+    ...parseSurvey(surveyDetailResponse),
+    intro: parseSurveyQuestion(surveyIntro),
+    outro: parseSurveyQuestion(surveyOutro),
+    questions: sortRecords(questions),
+  };
+};
+
+const parseSurveyQuestion = (questionResponse: any): SurveyQuestion => {
+  const parsedAnswer = (<object[]>questionResponse.answers)?.map((answer) =>
+    parseSurveyAnswer(answer)
+  );
+
+  return {
+    id: questionResponse.id,
+    displayOrder: questionResponse.displayOrder,
+    coverImageUrl: `${questionResponse.coverImageUrl}l`,
+    displayType: questionResponse.displayType,
+    text: questionResponse.text,
+    pick: questionResponse.pick,
+    answers: sortRecords(parsedAnswer),
+  };
+};
+
+const parseSurveyAnswer = (answerResponse: any): SurveyAnswer => ({
+  id: answerResponse.id,
+  displayOrder: answerResponse.displayOrder,
+  text: answerResponse.text,
 });
+
+const extractQuestionType = (
+  questions: SurveyQuestion[],
+  displayType: QuestionType
+): [
+  extractedQuestions: SurveyQuestion[],
+  remainingQuestions: SurveyQuestion[]
+] => partition(questions, (question) => question.displayType == displayType);
+
+const sortRecords = (records: object) =>
+  sortBy(records, [({ displayOrder }) => displayOrder]);
```

Comment:
```suggestion
const sortRecords = (records: object) =>
  records.sort((a, b) => b.displayOrder - a.displayOrder);
```

Does it work with that? If yes, it means you might not need lodash at all 😇